### PR TITLE
Time select component

### PR DESCRIPTION
**Pull request by shackpank**
Extracting this before it gets duplicated into anymore internal projects.

tqmoswr

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## Developing
+
+Clone the repo and `npm install`.
+
+`npm start` will create and watchify an example which you can open in your browser, at `doc/example.html`
+
+`npm test` for the unit tests.
+
+`npm lint` checks the code against our [guidelines](https://github.com/holidayextras/culture/blob/master/.eslintrc)
```
Comment (hxpaul) [inline README.md:30]: Oh I see you do use `make-up` for this so can remove and .gitignore the .eslintrc file

File: package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "react-time-select",
+  "version": "1.0.0",
+  "description": "Component to generate a dropdown list populated with a configurable time range",
+  "main": "index.js",
+  "scripts": {
+    "start": "ulimit -n 9999 && watchify -t babelify doc/example.js -o doc/example-built.js",
+    "test": "mocha --compilers js:babel/register --require intl test/test*",
```
Comment (hxpaul) [inline package.json:9]: if test also runs npm run lint it will make the CI easier I think

File: src/TimeSelect.jsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+var React = require('react');
+var Input = require('react-bootstrap').Input;
```
Comment (hxpaul) [inline src/TimeSelect.jsx:4]: should this be `UIToolkit.Input` instead? So it automatically adds on our tracking and things?

File: test/testTimeSelect.jsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  return component.props.children.map(function(option) {
+    return option.props.value;
+  });
+};
+
+var shallowRender = function(jsx) {
+  var shallowRenderer = TestUtils.createRenderer();
+  shallowRenderer.render(jsx);
+  return shallowRenderer.getRenderOutput();
+};
```
Comment (hxpaul) [inline test/testTimeSelect.jsx:28]: good call, and reusable, can we share these little helpers somewhere?

File: src/TimeSelect.jsx (hunk 3 above)
Comment (shackpank) [inline src/TimeSelect.jsx:4]: `UIToolkit` does not export a `select` type, I was recommended to stick with react-bootstrap as we didn't have a good reason for building that.

File: README.md (hunk 1 above)
Comment (shackpank) [inline README.md:30]: I did wonder, but nobody informed was around to ask! Will get it ignored.

File: package.json (hunk 2 above)
Comment (shackpank) [inline package.json:9]: I'd prefer not to mix up those concerns. Don't mind adding a `make ci` to combine them, but I don't think `npm test` should do things unrelated to testing.

Comment (hx-markterry) [inline package.json:9]: `npm run ci` would be good, but as its only two commands the current travis config is trivial

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+language: node_js
+node_js:
+  - "0.12"
```
Comment (hx-markterry) [inline .travis.yml:4]: is it worth testing 0.12, or maybe instead test for v4 which is closer to the v5 extended life launch?

File: doc/example.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+var React = require('react');
```
Comment (hx-markterry) [inline doc/example.js:4]: should we lint the example?

File: index.js [hunk 7]
```diff
@@ -0,0 +1 @@
+module.exports = require('./src/TimeSelect.jsx');
```
Comment (hx-markterry) [inline index.js]: can we ditch this and just set the `main` property in the `package.json`?

Comment (hx-markterry) [inline index.js]: Also its strange how this passes linting even though it has no `use strict` line.

**Comment by hx-markterry** [comment]
Need to remove https://github.com/shackpank/react-time-select/blob/master/.eslintrc from the source branch

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+language: node_js
```
Comment (hx-markterry) [inline .travis.yml:1]: Could we add `sudo: false` and cache the node_modules to give a speed up on builds.

File: src/TimeSelect.jsx [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+var TimePicker = React.createClass({
+  mixins: [ ReactIntl.IntlMixin ],
+
+  propTypes: {
+    className: React.PropTypes.string,
+    label: React.PropTypes.string,
+    name: React.PropTypes.string,
+    value: React.PropTypes.instanceOf(Date),
+    start: React.PropTypes.number,
```
Comment (hx-markterry) [inline src/TimeSelect.jsx:15]: there so needs to be a react time prop type validator, passing time as an integer doesn't quite fit.

There is a date validator available so we could spend an amount of time and extend this project: https://www.npmjs.com/package/react-validator-prop-types

**Comment by hx-markterry** [comment]
looking good, only a few minor comments

File: index.js (hunk 7 above)
Comment (hxpaul) [inline index.js]: I've noticed linting only requires `use strict` if your code has a function in it

File: .travis.yml (hunk 5 above)
Comment (shackpank) [inline .travis.yml:4]: `jsdom` deprecated everything below `iojs` in one of its releases, we can't support both groups of versions simultaneously

File: package.json [hunk 10]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "react-time-select",
+  "version": "1.0.0",
+  "description": "Component to generate a dropdown list populated with a configurable time range",
+  "main": "index.js",
+  "scripts": {
+    "start": "ulimit -n 9999 && watchify -t babelify doc/example.js -o doc/example-built.js",
```
Comment (lukehansell) [inline package.json]: `ulimit -n 9999` should go in`prestart`:

```
{
  "prestart": "ulimit -n 9999",
  "start": "watchify ... "
}
```

File: .travis.yml (hunk 8 above)
Comment (shackpank) [inline .travis.yml:1]: Done

File: doc/example.js (hunk 6 above)
Comment (shackpank) [inline doc/example.js:4]: Done, but is there a way to have make-up lint just the file? I've had to add an `.eslintignore` to stop it finding 15k linting errors in the browserified build

**Comment by shackpank** [comment]
@hx-markterry the eslintrc is removed - PR here is from [this fork](https://github.com/shackpank/react-time-select-1), not that repo

**Comment by hx-markterry** [comment]
ok with me :ok:

**Pull request merged**